pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -431,9 +436,19 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 			}
 		}
 		workload.SetAdmissionCheckState(&wlPatch.Status.AdmissionChecks, checkState)
+		message := fmt.Sprintf("Admission check %s set state %s", checkState.Name, checkState.State)
+		if checkState.Message != "" {
+			message += fmt.Sprintf(" with message %s", checkState.Message)
+		}
+		recorderMessages = append(recorderMessages, message)
 	}
 	if updated {
-		return c.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName), client.ForceOwnership)
+		if err := c.client.Status().Patch(ctx, wlPatch, client.Apply, client.FieldOwner(ControllerName), client.ForceOwnership); err != nil {
+			return err
+		}
+		for i := range recorderMessages {
+			c.record.Eventf(wl, corev1.EventTypeNormal, "WorkloadAdmissionChecksUpdated", api.TruncateEventMessage(recorderMessages[i]))
```

Comment:
```suggestion
			c.record.Eventf(wl, corev1.EventTypeNormal, "AdmissionCheckUpdated", api.TruncateEventMessage(recorderMessages[i]))
```

No need to repeat Workload, as this is already referencing a workload object.